python/lsst/daf/butler/_butler.py

```diff
@@ -457,7 +457,7 @@ def makeRepo(
             # branch, _everything_ in the config is expanded, so there's no
             # need to special case this.
             Config.updateParameters(RegistryConfig, config, full, toMerge=("managers",), overwrite=False)
-        configURI: Union[str, ButlerURI]
+        configURI: Union[str, ResourcePath]
```

Comment:
Best to use `ResourcePathExpression` for this (it'slike this `Union`, but also includes `pathlib.Path`).

Reply:
Okay. That is passed to dumpToUri which does take ResourcePathExpression.